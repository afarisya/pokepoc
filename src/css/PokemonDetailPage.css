.pokemon-detail-first-col {
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: space-between;
    min-height: 400px;
    padding: 20px 40px;
    background-color: rgba(0,0,0,0.1);
}

.pokemon-detail-second-col {
    padding: 20px 40px;
    background-color: #fff;
}

.pokemon-pictures-carousel {
    width: 100%;
}
.pokemon-pictures-carousel img {
    width: 200px;
    max-width: 100%;
}

#pokemon-detail-name {
    margin-bottom: 20px;
}

.catch-btn {
    margin-bottom: 20px;
    width: 125px;
    max-width: 100%;
}

.catched-nickname {
    margin-bottom: 20px;
}

#catch-pokemon-modal {
    text-align: center;
}

#catch-pokemon-modal .modal-body h5,
#catch-pokemon-modal .modal-body div, 
#catch-pokemon-modal .modal-body input {
    margin-bottom: 15px;
}

#catch-pokemon-modal button {
    margin: 0 10px 5px;
}

.pokemon-detail-types,
.pokemon-detail-moves {
    margin-bottom: 20px;
}

.pokemon-detail-types .badge,
.pokemon-detail-moves .badge {
    margin-left: 3px;
    margin-right: 3px;
    margin-bottom: 5px;
    font-size: 14px;
    padding-bottom: 6px !important;
}