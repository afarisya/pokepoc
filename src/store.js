import { createStore, applyMiddleware, compose } from 'redux';
import { routerMiddleware } from 'react-router-redux';
import thunk from 'redux-thunk';
// import WebsocketMiddleware from './middleware/WebsocketMiddleware';
import { createBrowserHistory, createHashHistory } from 'history';
import rootReducer from './reducers';

export const history = createBrowserHistory()

// Use this if you want to use routing with hash
// export const history = createHashHistory();

const initialState = {}
const enhancers = []
const middleware = [
  thunk,
  // logger,
  routerMiddleware(history)//,
//   WebsocketMiddleware
]

if (process.env.NODE_ENV === 'development') {
  const devToolsExtension = window.devToolsExtension

  if (typeof devToolsExtension === 'function') {
    enhancers.push(devToolsExtension())
  }
}

const composedEnhancers = compose(
  applyMiddleware(...middleware),
  ...enhancers
)

const store = createStore(
  rootReducer,
  initialState,
  composedEnhancers
)

export default store