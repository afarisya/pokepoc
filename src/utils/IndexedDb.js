// import store from '../store'
// import { setIndexedDB } from "../reducers/GlobalReducers";
// import { setUserPicture } from "../reducers/ProfileReducers";
// import { sndAudioStream, reqInitSendBlobChange, sndEosStream, sndEndStream } from '../reducers/NewMeetingReducers';

// import defaultPicture from '../img/admin.png';


export var db;

// IndexedDB
var indexedDB = window.indexedDB || window.webkitIndexedDB || window.mozIndexedDB || window.OIndexedDB || window.msIndexedDB;
// var IDBTransaction = window.IDBTransaction || window.webkitIDBTransaction || window.OIDBTransaction || window.msIDBTransaction;
var dbVersion = 1.0;

// Create/open database
var request = indexedDB.open("MyPokemonList", dbVersion);

var lastIndex = null;

request.onsuccess = function (event) {
    console.log("Success creating/accessing IndexedDB database");

    db = request.result;

    db.onerror = function (event) {
        console.log("Error creating/accessing IndexedDB database");
    };
    
}

request.onerror = function (event) {
    console.log("Error creating/accessing IndexedDB database");
};

// For future use. Currently only in latest Firefox versions
request.onupgradeneeded = function (event) {
    // console.log("Upgrading IndexedDb");
    var dataBase = event.target.result;

    var store = dataBase.createObjectStore("pokemons", { 
                    keyPath: 'pokemonId'//,
                    // autoIncrement: true 
                });
    store.createIndex("pokemonId", "pokemonId", { unique: true });
    store.createIndex("pokemonName", "pokemonName", { unique: false });
    store.createIndex("pokemonNickname", "pokemonNickname", { unique: false });

};

export var createObjectStore = function (dataBase, objStoreName) {
    // Create an objectStore
    // console.log("Creating objectStore")
    dataBase.createObjectStore(objStoreName);
};

export var getMyPokemonList = function(offset, limit){
    return new Promise((resolve, reject) => {
        var objectStore = db.transaction("pokemons").objectStore("pokemons");
        var pokemons = [];

        var iterate = 0;

        var request = objectStore.openCursor();

        request.onsuccess = function(event) {
            var cursor = event.target.result;
            if (cursor) {
                if ( iterate >= offset && pokemons.length < limit ) {
                    pokemons.push(cursor.value);
                }

                console.log(pokemons)
                if ( pokemons.length === limit ) {
                    getMyPokemonCount()
                        .then((count) => {
                            resolve(
                                {
                                    count: iterate,
                                    pokemons: pokemons
                                }
                            );
                        })
                        .catch((err) => {
                            reject("error")
                        })
                }

                iterate+=1;
                cursor.continue();

            } else {
                getMyPokemonCount()
                    .then((count) => {
                        resolve(
                            {
                                count: iterate,
                                pokemons: pokemons
                            }
                        );
                    })
                    .catch((err) => {
                        reject("error")
                    })
            }
        }

        request.onerror = function(event) {
            reject("error")
        }
    })


    // IDBKeyRange.bound(0, N, false, true);
}

export var getMyPokemonCount = function(){
    return new Promise((resolve, reject) => {
        var transaction = db.transaction(['pokemons'], 'readonly');
        var objectStore = transaction.objectStore('pokemons');
            
        var countRequest = objectStore.count();

        countRequest.onsuccess = function() {
            console.log(countRequest.result)
            resolve(countRequest.result);
        }

        countRequest.onerror = function(event) {
            reject("error");
        };
    })
}

export var getPokemonById = function(id){
    return new Promise((resolve, reject) => {
        var transaction = db.transaction(["pokemons"]);
        var objectStore = transaction.objectStore("pokemons");
        var request = objectStore.get(id);
        
        request.onsuccess = function(event) {
            // Do something with the request.result!
            resolve(request.result);
        };

        request.onerror = function(event) {
            reject("error");
        };
    })
}

export var addPokemon = function(obj){    
    return new Promise((resolve, reject) => {
        var request = db.transaction(["pokemons"], "readwrite")
                    .objectStore("pokemons")
                    .add(obj);

        // transaction.objectStore(objStoreName).put(obj, key);

        request.onsuccess = function(event) {
            resolve("success");
        };

        request.onerror = function(event) {
            reject("error");
        };
    })

};

export var releasePokemon = function(pokemonId){
    return new Promise((resolve, reject) => {
        var request = db.transaction(["pokemons"], "readwrite")
                    .objectStore("pokemons")
                    .delete(pokemonId);

        request.onsuccess = function(event) {
            resolve("success");
        };

        request.onerror = function(event) {
            reject("error");
        };
    })
}