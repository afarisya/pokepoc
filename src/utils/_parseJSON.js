export function _parseJSON(response) {
    return response.text()
        .then( text => {
            return text ? JSON.parse(text) : {}
    })
}