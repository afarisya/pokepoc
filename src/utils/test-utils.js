import { ConnectedRouter } from 'react-router-redux';
import { mount } from 'enzyme';
import { mergeDeepRight } from 'ramda';
import React from 'react';
import { Provider } from 'react-redux';
import { createStoreWithMiddleWare, history } from '../store';
import rootReducer from '../reducers';

export const mockData = {
  pokemon: {
    id: 1,
    species: { name: 'bulbasaur' },
    sprites: {
      front_default:
        'https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/26.png'
    }
  }
};

export const makeMountRender = (Component, defaultProps = {}) => {
  return (customProps = {}) => {
    const props = {
      ...defaultProps,
      ...customProps
    };
    return mount(<Component {...props} />);
  };
};

export const makeStore = (customState = {}) => {
  const root = rootReducer({}, { type: '@@INIT' });
  const state = mergeDeepRight(root, customState);

  return createStoreWithMiddleWare(rootReducer, state);
};

export const reduxify = (Component, props = {}, state = {}) => {
  return function reduxWrap() {
    return (
      <Provider store={makeStore(state)}>
        <ConnectedRouter history={history}>
          <Component {...props} />
        </ConnectedRouter>
      </Provider>
    );
  };
};

export const snapshotify = reactWrapper => {
  return reactWrapper.html();
};

export const mocker = apiMock => ({
  fetchRandomPokemon() {
    apiMock
      .onGet(/https:\/\/pokeapi.co\/api\/v2\/pokemon\/\d+/)
      .reply(config => {
        const numberCheck = RegExp(/pokemon\/(\d+)/);
        const pokemonInt = Number(numberCheck.exec(config.url)[1]);

        if (pokemonInt > 151) {
          return [500];
        }

        return [200, mockData.pokemon];
      });

    return this;
  }
});

export const ticks = (callbacks = []) => {
  if (callbacks.length < 1) {
    return null;
  }

  setTimeout(() => {
    const cb = callbacks.shift();

    if (cb) {
      cb();
    }

    ticks(callbacks);
  });
};
