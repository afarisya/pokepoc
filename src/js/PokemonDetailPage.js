import $ from 'jquery';
import React from 'react';
import { connect } from 'react-redux';
import { 
    Container, 
    Row, 
    Col, 
    Modal, 
    ModalBody, 
    Button,
    Input,
    Carousel,
    CarouselIndicators,
    CarouselItem,
    CarouselControl,
    Badge
} from 'reactstrap';
import store from '../store';

import { 
    reqPokemonIdChange, 
    reqPokemonDetail, 
    reqPokemonPictures,
    rcvPokemonCatched,
    clearPokemonDetail
} from '../reducers/PokemonDetailReducers';
import { addPokemon } from '../utils/IndexedDb';

// CSS
import '../css/PokemonDetailPage.css'


class PokemonDetailPage extends React.Component {    
    constructor(props) {
        super(props);
        this.state = {
            modal           : false,    // modal
            activeIndex     : 0,        // carousel 
            items           : [],       // carousel pictures

            inputNickname   : "",
            tryToCatch      : null
        }
        this.next = this.next.bind(this);
        this.previous = this.previous.bind(this);
        this.goToIndex = this.goToIndex.bind(this);
        this.onExiting = this.onExiting.bind(this);
        this.onExited = this.onExited.bind(this);
        this.modalToggle = this.modalToggle.bind(this);
    }

    componentWillMount() {
        var pokemon_id = this.props.match.params.pokemonId;
        
        store.dispatch(reqPokemonIdChange(pokemon_id));
        store.dispatch(reqPokemonDetail(pokemon_id));
        store.dispatch(reqPokemonPictures(pokemon_id));
    }
    
    componentDidMount() {
        this.renderData(this.props);
    }

    componentWillReceiveProps(nextProps) {
        if (
            nextProps.id === this.props.id && 
            nextProps.pictures !== this.props.pictures 
        ) {
            this.renderData(nextProps);
        }
    }

    componentWillUnmount() {
        store.dispatch(clearPokemonDetail());
    }

    renderData = (props) => {
        var items = [];

        if ( props.pictures.front_default !== null ) {
            items.push({
                key: 1,
                src: props.pictures.front_default,
                altText: ''
            });
        }

        if ( props.pictures.back_default !== null ) {
            items.push({
                key: 2,
                src: props.pictures.back_default,
                altText: ''
            });
        }

        this.setState({
            items: items
        }) 

    }

    modalToggle() {
        this.setState({
            modal: !this.state.modal
        }, () => {
            if ( this.state.modal ) {
                setTimeout(() => {
                    $("#nickname-input").focus();
                }, 50);
            }
        });
    }

    onExiting() {
        this.animating = true;
    }

    onExited() {
        this.animating = false;
    }
    
    next() {
        if (this.animating) return;
        const nextIndex = this.state.activeIndex === this.state.items.length - 1 ? 0 : this.state.activeIndex + 1;
        this.setState({ activeIndex: nextIndex });
    }
    
    previous() {
        if (this.animating) return;
        const nextIndex = this.state.activeIndex === 0 ? this.state.items.length - 1 : this.state.activeIndex - 1;
        this.setState({ activeIndex: nextIndex });
    }
    
    goToIndex(newIndex) {
        if (this.animating) return;
        this.setState({ activeIndex: newIndex });
    }

    catchPokemon = () => {
        var try_to_catch = Math.floor(Math.random() * Math.floor(2));

        this.setState({
            tryToCatch: try_to_catch
        }, () => {
            this.modalToggle();
        })
    }

    
    recatchPokemon = () => {
        this.modalToggle();
        this.catchPokemon();
    }

    savePokemon = () => {
        // Add to MyPokemonList
        addPokemon({
            pokemonId: this.props.id,
            pokemonName: this.props.name,
            pokemonNickname: this.state.inputNickname
        })
            .then(data => {
                store.dispatch(rcvPokemonCatched(true, this.state.inputNickname));    

                this.setState({
                    inputNickname: ""
                })         
                this.modalToggle();   
            })
            .catch(err => {
                // console.log(err)
            });
    }

    nicknameOnChange = (e) => {
        this.setState({
            inputNickname: e.target.value
        })
    }

    render() {
        // console.log(this.state.items)
        

        return (
            <Container>
                <Row>
                    <Col className="pokemon-detail-first-col" xs="12" md="12">
                        <Carousel
                            className="pokemon-pictures-carousel"
                            activeIndex={this.state.activeIndex}
                            next={this.next}
                            previous={this.previous}
                        >
                            <CarouselIndicators 
                                items={this.state.items} 
                                activeIndex={this.state.activeIndex} 
                                onClickHandler={this.goToIndex} 
                            />
                            {
                                this.state.items.map((item, index) => {
                                    return (
                                        <CarouselItem
                                            onExiting={this.onExiting}
                                            onExited={this.onExited}
                                            key={index}
                                        >
                                            <img src={item.src} alt={item.altText} />
                                        </CarouselItem>
                                    );
                                })
                            }
                            <CarouselControl direction="prev" directionText="Previous" onClickHandler={this.previous} />
                            <CarouselControl direction="next" directionText="Next" onClickHandler={this.next} />
                        </Carousel>
                        <h2 id="pokemon-detail-name">{this.props.name}</h2>
                        {
                            !this.props.catched ?
                                <React.Fragment>
                                    <Button className="catch-btn" color="primary" onClick={this.catchPokemon}>Catch</Button>
                                    { this.state.tryToCatch === 1 ?
                                        <Modal 
                                            isOpen={this.state.modal} 
                                            toggle={this.modalToggle}
                                            backdrop="static" 
                                            id="catch-pokemon-modal"
                                        >
                                            <ModalBody>
                                                <h5>Pokemon <span style={{fontWeight: "bold"}}>{this.props.name}</span> catched!</h5>
                                                <div>Give it a nickname</div>
                                                <Input id="nickname-input" value={this.state.inputNickname} onChange={this.nicknameOnChange.bind(this)}/>
                                                <Button 
                                                    color="secondary"
                                                    onClick={this.modalToggle}
                                                >
                                                    Cancel
                                                </Button>
                                                <Button 
                                                    color="primary"
                                                    onClick={this.savePokemon}
                                                >
                                                    Save
                                                </Button>
                                            </ModalBody>
                                        </Modal>
                                        : null
                                    }
                                    { this.state.tryToCatch === 0 ?
                                        <Modal 
                                            isOpen={this.state.modal} 
                                            toggle={this.modalToggle}
                                            backdrop="static" 
                                            id="catch-pokemon-modal"
                                        >
                                            <ModalBody>
                                                <h5>Failed to catch pokemon <span style={{fontWeight: "bold"}}>{this.props.name}</span>!</h5>
                                                <Button 
                                                    color="secondary"
                                                    onClick={this.modalToggle}
                                                >
                                                    Cancel
                                                </Button>
                                                <Button 
                                                    color="primary"
                                                    onClick={this.recatchPokemon}
                                                >
                                                    Try Again
                                                </Button>
                                            </ModalBody>
                                        </Modal>
                                        : null
                                    }
                                </React.Fragment>
                                :
                                <React.Fragment>
                                    <Button className="catch-btn" color="secondary" disabled>Catched!</Button>
                                    <div className="catched-nickname">Your {this.props.name} nickname is <span style={{fontWeight: "700"}}>{this.props.nickname}</span></div>
                                </React.Fragment>
                        }
                    </Col>
                    <Col className="pokemon-detail-second-col" xs="12" md="12">
                        <div className="pokemon-detail-types">
                            <h5>Types</h5>
                            {
                                this.props.types.map((type, index) => {
                                    return (
                                        <Badge color="danger" key={index}>{type.type.name}</Badge>
                                    )
                                })
                            }
                        </div>
                        <div className="pokemon-detail-moves">
                            <h5>Moves</h5>
                            {
                                this.props.moves.map((move, index) => {
                                    return (
                                        <Badge color="secondary" key={index}>{move.move.name}</Badge>
                                    )
                                })
                            }
                        </div>
                    </Col>
                    <Col xs="12">
                    </Col>
                </Row>
            </Container>
        );
    }
};

function mapStateToProps(state) {
	return {
		id          : state.PokemonDetailReducers.id,
		name        : state.PokemonDetailReducers.name,
		pictures    : state.PokemonDetailReducers.pictures,
		moves       : state.PokemonDetailReducers.moves,
		types       : state.PokemonDetailReducers.types,
		catched     : state.PokemonDetailReducers.catched,
		nickname    : state.PokemonDetailReducers.nickname,
	}
}

export default connect(mapStateToProps)(PokemonDetailPage);