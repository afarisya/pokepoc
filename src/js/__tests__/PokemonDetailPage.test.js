// PokemonDetailPage.test.js

import React from 'react';
import { shallow } from 'enzyme';
import { createIndexedDB } from '../../utils/IndexedDb';
import { history } from '../../store'
// require("fake-indexeddb/auto");

import PokemonDetailPage from '../PokemonDetailPage';


describe("PokemonDetailPage", () => {
    const page = shallow(
        <PokemonDetailPage.WrappedComponent 
            id="bulbasaur"
            name="bulbasaur"
            pictures={[]}
            moves={[]}
            types={[]}
            catched={false}
            nickname=""
            match={ {params: "bulbasaur"} }
        />
    );

    it("should render pokemon detail page if not catched", () => {   
        expect(page.getElements()).toMatchSnapshot();
    });

    it("should show text 'bulbasaur' in #pokemon-detail-name", () => { 
        expect(page.find('#pokemon-detail-name').text()).toEqual("bulbasaur");
    });

    const page2 = shallow(
        <PokemonDetailPage.WrappedComponent 
            id="bulbasaur"
            name="bulbasaur"
            pictures={[]}
            moves={[]}
            types={[]}
            catched={true}
            nickname="bulb"
            match={ {params: "bulbasaur"} }
        />
    );

    it("should render pokemon detail page if catched", () => {   
        expect(page.getElements()).toMatchSnapshot();
    });

    // it("should show 1 text 'Catched!' in #catch-btn if catched = true", () => { 
    //     expect(page2.find('button').at(1).text()).toEqual("Catched!");
    // });

    // it("should show PokemonPagination on pokemon detail page if pokemons.length = 1", () => {  
    //     expect(page2.find('PokemonPagination').length).toEqual(1);
    // });

    // expect(page.prop('activePage')).toEqual(1);
});