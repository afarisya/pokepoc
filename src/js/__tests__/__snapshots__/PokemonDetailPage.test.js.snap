// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonDetailPage should render pokemon detail page if catched 1`] = `
Array [
  <Container
    tag="div"
  >
    <Row
      tag="div"
    >
      <Col
        className="pokemon-detail-first-col"
        md="12"
        tag="div"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
        xs="12"
      >
        <Carousel
          activeIndex={0}
          className="pokemon-pictures-carousel"
          interval={5000}
          keyboard={true}
          next={[Function]}
          pause="hover"
          previous={[Function]}
          slide={true}
        >
          <CarouselIndicators
            activeIndex={0}
            items={
              Array [
                Object {
                  "altText": "",
                  "key": 1,
                  "src": undefined,
                },
                Object {
                  "altText": "",
                  "key": 2,
                  "src": undefined,
                },
              ]
            }
            onClickHandler={[Function]}
          />
          <CarouselItem
            appear={false}
            enter={true}
            exit={true}
            in={false}
            mountOnEnter={false}
            onEnter={[Function]}
            onEntered={[Function]}
            onEntering={[Function]}
            onExit={[Function]}
            onExited={[Function]}
            onExiting={[Function]}
            slide={true}
            tag="div"
            timeout={600}
            unmountOnExit={false}
          >
            <img
              alt=""
            />
          </CarouselItem>
          <CarouselItem
            appear={false}
            enter={true}
            exit={true}
            in={false}
            mountOnEnter={false}
            onEnter={[Function]}
            onEntered={[Function]}
            onEntering={[Function]}
            onExit={[Function]}
            onExited={[Function]}
            onExiting={[Function]}
            slide={true}
            tag="div"
            timeout={600}
            unmountOnExit={false}
          >
            <img
              alt=""
            />
          </CarouselItem>
          <CarouselControl
            direction="prev"
            directionText="Previous"
            onClickHandler={[Function]}
          />
          <CarouselControl
            direction="next"
            directionText="Next"
            onClickHandler={[Function]}
          />
        </Carousel>
        <h2
          id="pokemon-detail-name"
        >
          bulbasaur
        </h2>
        <React.Fragment>
          <Button
            className="catch-btn"
            color="primary"
            onClick={[Function]}
            tag="button"
          >
            Catch
          </Button>
        </React.Fragment>
      </Col>
      <Col
        className="pokemon-detail-second-col"
        md="12"
        tag="div"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
        xs="12"
      >
        <div
          className="pokemon-detail-types"
        >
          <h5>
            Types
          </h5>
        </div>
        <div
          className="pokemon-detail-moves"
        >
          <h5>
            Moves
          </h5>
        </div>
      </Col>
      <Col
        tag="div"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
        xs="12"
      />
    </Row>
  </Container>,
]
`;

exports[`PokemonDetailPage should render pokemon detail page if not catched 1`] = `
Array [
  <Container
    tag="div"
  >
    <Row
      tag="div"
    >
      <Col
        className="pokemon-detail-first-col"
        md="12"
        tag="div"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
        xs="12"
      >
        <Carousel
          activeIndex={0}
          className="pokemon-pictures-carousel"
          interval={5000}
          keyboard={true}
          next={[Function]}
          pause="hover"
          previous={[Function]}
          slide={true}
        >
          <CarouselIndicators
            activeIndex={0}
            items={
              Array [
                Object {
                  "altText": "",
                  "key": 1,
                  "src": undefined,
                },
                Object {
                  "altText": "",
                  "key": 2,
                  "src": undefined,
                },
              ]
            }
            onClickHandler={[Function]}
          />
          <CarouselItem
            appear={false}
            enter={true}
            exit={true}
            in={false}
            mountOnEnter={false}
            onEnter={[Function]}
            onEntered={[Function]}
            onEntering={[Function]}
            onExit={[Function]}
            onExited={[Function]}
            onExiting={[Function]}
            slide={true}
            tag="div"
            timeout={600}
            unmountOnExit={false}
          >
            <img
              alt=""
            />
          </CarouselItem>
          <CarouselItem
            appear={false}
            enter={true}
            exit={true}
            in={false}
            mountOnEnter={false}
            onEnter={[Function]}
            onEntered={[Function]}
            onEntering={[Function]}
            onExit={[Function]}
            onExited={[Function]}
            onExiting={[Function]}
            slide={true}
            tag="div"
            timeout={600}
            unmountOnExit={false}
          >
            <img
              alt=""
            />
          </CarouselItem>
          <CarouselControl
            direction="prev"
            directionText="Previous"
            onClickHandler={[Function]}
          />
          <CarouselControl
            direction="next"
            directionText="Next"
            onClickHandler={[Function]}
          />
        </Carousel>
        <h2
          id="pokemon-detail-name"
        >
          bulbasaur
        </h2>
        <React.Fragment>
          <Button
            className="catch-btn"
            color="primary"
            onClick={[Function]}
            tag="button"
          >
            Catch
          </Button>
        </React.Fragment>
      </Col>
      <Col
        className="pokemon-detail-second-col"
        md="12"
        tag="div"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
        xs="12"
      >
        <div
          className="pokemon-detail-types"
        >
          <h5>
            Types
          </h5>
        </div>
        <div
          className="pokemon-detail-moves"
        >
          <h5>
            Moves
          </h5>
        </div>
      </Col>
      <Col
        tag="div"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
        xs="12"
      />
    </Row>
  </Container>,
]
`;
